修复headless中字段查询及多轮对话使用问题

BigdataQIJI opened the pull request:
一、 修复生成的SQL字段不在数据集内，导致查询异常的问题，由原来的字段数量判断更改为查询字段是否包含维度和度量字段
#2007
二、 修复开启多轮对话会使用默认的gpt-4o-mini账号去进行过程检测而不是使用配置好的本地大模型的问题
#2251
问题说明：
1、问题一中的原有方法中是根据查询字段数量是否大于维度+度量的字段数量来判断是否包含不匹配的字段，这样即使text2sql是对的，因大模型自己新增了1个或多个字段也会报错，很影响体验。现更改为查询字段是否包含维度+度量的字段来判断是否包含不匹配的字段。如大模型自行新增字段，可通过修改提示词模板来约束新增字段问题。

Commit:
(fix)(headless)修复生成的SQL字段不在数据集内，导致查询异常的问题，由原来的字段数量判断更改为查询字段是否包含维度和度量字段

Commit:
(fix)(headless) <修复开启多轮对话会使用默认的gpt-4o-mini账号去进行过程检测而不是使用配置好的本地大模型的问题>

hurentian commented:
您好，第一点的修改产生了新的问题
维度值导入字典时，进来的queryFields总是为英文字段名，但您修改的ontologyMetricsDimensions这个Set中总为中文名，导致必然会出现以下异常。
![image](https://github.com/user-attachments/assets/9e8e5c42-b526-4ad0-870a-d197df9eca81)
![image](https://github.com/user-attachments/assets/d73683f9-929c-4853-b1b5-fa64c6621c4f)
页面路径为：语意建模->选择数据域->模型管理->点击模型名称进入一个模型->维度管理->维度值设置->维度值可见->立即导入字典

请问下这个改如何修复呢？

BigdataQIJI committed:
修复维度值导入字典时，维度值和度量值为英文名导致字段不匹配的问题

修复维度值导入字典时，维度值和度量值为英文名导致字段不匹配的问题。
维度值导入字典时会遍历调用到SqlQueryParser方法，这时的维度值或度量值为Bizname而非name，新增一个set用于添加Bizname，并在判断中新加入Bizname名的判断，只要两个条件符合一个即可

BigdataQIJI commented:
> 您好，第一点的修改产生了新的问题 维度值导入字典时，进来的queryFields总是为英文字段名，但您修改的ontologyMetricsDimensions这个Set中总为中文名，导致必然会出现以下异常。 ![image](https://private-user-images.githubusercontent.com/38875515/443519262-9e8e5c42-b526-4ad0-870a-d197df9eca81.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcyMTkzMTQsIm5iZiI6MTc0NzIxOTAxNCwicGF0aCI6Ii8zODg3NTUxNS80NDM1MTkyNjItOWU4ZTVjNDItYjUyNi00YWQwLTg3MGEtZDE5N2RmOWVjYTgxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTE0VDEwMzY1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1NDAxNTk1YTFjN2YxNjljNDQzYzI5MzdlMmE2YTA4ZDRiNmU4YWQ1ZGUwMjQxNDg5Mjk4YTgzYTI1YTUzYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4lrootHcHEYhGpPhU8I9SaWewET9UGY_wX1F6kD-jaQ) ![image](https://private-user-images.githubusercontent.com/38875515/443519883-d73683f9-929c-4853-b1b5-fa64c6621c4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcyMTkzMTQsIm5iZiI6MTc0NzIxOTAxNCwicGF0aCI6Ii8zODg3NTUxNS80NDM1MTk4ODMtZDczNjgzZjktOTI5Yy00ODUzLWIxYjUtZmE2NGM2NjIxYzRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTE0VDEwMzY1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4YzlhMWYyYTllMGE1ZDYyMjVkMWE1Njk4NmVhZDIwOGJhMGFiOGUyNjhiYmY5NWY1ZDkxNDIxZDdjMGM1MjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dicuofAnc5bYQrnbbzQRpjQ7RY88Ciz3uV4gffzbIPg) 页面路径为：语意建模->选择数据域->模型管理->点击模型名称进入一个模型->维度管理->维度值设置->维度值可见->立即导入字典
> 
> 请问下这个改如何修复呢？

您好，这个问题已修复，抱歉我忽略了导入维度值的操作，新增加了对Bizname的匹配条件，二个条件满足一个即可。

BigdataQIJI committed:
Merge branch 'tencentmusic:master' into master_latest

jerryjzhang commented:
谢谢！
